### DISCUSSION TITLE
The pattern for face numbers for element and neighbor

### DISCUSSION THREAD

**Discussion by ambehnam** [idaholab/moose#21697] [Q&A General] [answered]
Hello there. I have a question regarding the face numbers for element and neighbor. What pattern MOOSE/Libmesh uses to define _neighbor_side and _current_side? 

From the HEX8 element, the local node numbers are defined in the documentation. However, I couldn't find how the faces are numbered and set for _neighbor_side and _current_side.

I have a 2 element model with an interface between them. The _current_side value is 2, and I expected the _neighbor_side to be 1, so  the two will be back to back and have opposite faces. But the _neighbor_side value is 4. What is the reason for this?

Thanks a lot,
Amir

**Comment by GiudGiud** [comment, marked as answer]
Hello

I looked at the doxygen and the answer is here:
https://libmesh.github.io/doxygen/classlibMesh_1_1Hex8.html#ae6fd20a71114610dac6b527498248007

side 1 is made of nodes {0, 3, 2, 1}, 
side 2 of nodes  {0, 1, 5, 4},
etc

This is HEX8, it ll be different for each element type

Guillaume
